src/docs/patterns/forms.mdx

```diff
@@ -34,3 +37,45 @@ If your form fields have complex requirements for valid values, you can list the
 If there is an error with the data entered into a field, the field should indicate to the user that there is a problem. Label and helper text should be changed to a red color and a warning icon should appear in the text input. The error message should be displayed in the helper text location below the field (any existing helper text should be replaced with the error message).
 
 Error messages should be succinct and should not provide information that could compromise security (e.g., "Invalid Credentials" instead of "Incorrect Password").
+
+In general, there are two different types of input field verification: on blur, on change, on page change.
+
+### Verify Input on Blur
+
+In order to not "push" the user too much, most text fields should remain "verify on blur". That is, most of the time, you should only verify user input when they stop typing.
+
+<ImageGrid
+    images={[OnBlurImage]}
+    gridImageProps={{ sm: 12 }}
+    caption={[
+        '1: Do not check for validity or display error messages when the user is typing the desired text.',
+        '2: The user is finished and defocuses / “blurs away” from the input field. Now their input should be checked.',
+    ]}
+/>
+
+### Verify Input on Change
+
+When it comes to a password field, the validity check feedback should be displayed each time the user makes a change to the input field.
+
+<ImageGrid
+    images={[OnChangeImage]}
+    gridImageProps={{ sm: 12 }}
+    caption={[
+        '1: The criteria for the password shall be displayed interactively while the user is changing their input.',
+        '2: “Confirm Password” field should behave like a regular text field and do “verify input on blur”.',
+    ]}
+/>
```

Comment:
I don't think I agree with Confirm Password verifying onBlur. I think in most situations that's the last input in the form (and then there's some sort of continue button). If that continue button is enabled/disabled based on passwords matching, then the user has to do an extra click away from the field before they can click the continue button. Or if the continue button is always enabled, then if the passwords don't match they can click the button and then get presented with an error. It doesn't feel like the smoothest UX.
```suggestion
### Verifying on Change

Some fields should verify the user input as it changes. The most common case for this type of verification is a new password field (or a confirm password field). The verification feedback should be displayed / updated each time the user makes a change to the input field.

<ImageGrid
    images={[OnChangeImage]}
    gridImageProps={{ sm: 12 }}
    caption={[
        '1: The password criteria checks shall be updated as the user changes their input.',
        '2: “Confirm Password” field should check for a match with password to give immediate feedback for when the user can proceed.',
    ]}
/>
```

Reply:
Good point! Although Google does the "verifying manually" approach on the confirm password field for both creating a new account and change the account password workflow.

Reply:
Hmm, that's interesting. Maybe we wait and see if somebody from Studio Blue wants to weigh in on this and if it's somewhere that we want to deviate.

Reply:
Let me add a "when to enable / disable the submit button" section in the meantime